CMakeLists.txt

```diff
@@ -51,17 +58,26 @@ file(GLOB SOURCES "edlib/src/*.cpp")
 # Create libraries.
 add_library(edlib ${SOURCES})
 set_target_properties(edlib
-    PROPERTIES
+  PROPERTIES
+    CXX_VISIBILITY_PRESET hidden
+    VISIBILITY_INLINES_HIDDEN ON
     VERSION ${PROJECT_VERSION}
-    SOVERSION ${PROJECT_VERSION})
+    SOVERSION ${PROJECT_VERSION}
```

Comment:
I didn't change this line. But don't you want to set SOVERSION with major only? Don't you follow semver?

Reply:
Yes seems that there are tabs in CMakeLists.txt only (in `EDLIB_ENABLE_INSTALL` block specifically). I can replace those tabs in this PR if you want.

Reply:
Semver is usually for API, not ABI. There are many projects that break the ABI within a major version, yet keep the API compatible. I would tread carefully with tying the `SOVERSION` to `PROJECT_VERSION`.

Reply:
Hmm, how would you do deal with SOVERSION then @SoapZA ? Should it not also change the version when major version of the API changes? What is practically the soundest here, keeping in mind that I personally don't have very deep understanding of what means that ABI changed (I get the idea of what ABI is, but in practice, when would it change for edlib?).

Reply:
We should maybe look into adding a CI job that detects an ABI change. `libabigail` for instance is capable of detecting ABI changes, and then solution here would be to bump the `SOVERSION` on every ABI change. That is how libtool does it. Given that the interface of Edlib is rather thin, I don't anticipate major breaks often, but it's still important for packagers such as Debian and Gentoo.

Reply:
Sounds good! This is certainly out of scope of this PR, so I guess it is best if we create an issue for this. Would you like to create it, since you understand the whole thing better?